rdt/transformers/datetime.py

```diff
@@ -112,6 +117,9 @@ def _fit(self, data):
             data (pandas.Series):
                 Data to fit the transformer to.
         """
+        if self.datetime_format is None and data.dtype == 'object':
+            self.datetime_format = _guess_datetime_format_for_array(data.to_numpy())
```

Comment:
Should we use the public method that this uses instead? https://github.com/pandas-dev/pandas/blob/258cfccf8106814504ca07a118037d3f32073732/pandas/core/tools/datetimes.py#L131
They basically just filter to make sure they aren't using a `nan` value to check. I don't know if it is risky to use hidden methods